sycl/include/sycl/detail/pi.h

```diff
@@ -53,9 +53,12 @@
 // 10.14 Add PI_EXT_INTEL_DEVICE_INFO_FREE_MEMORY as an extension for
 // piDeviceGetInfo.
 // 11.15 piEventCreate creates even in the signalled state now.
+// 11.16 piSetEventProperty modifies properties of a pi_event.
+//       Currently only supports marking a to be constructed pi_event
+//       user-visible
```

Comment:
what is the definition of a "user-visible" here? what the not user0visible events and why would their handling be different?

Reply:
A user visible event is any event that has an  associated SYCL event and the end-user can interact with it. For example, an event return by queue.submit is user visible, but the command group submitted might have requirements which also have events associated with them, are not user visible is they're used internally by the runtime. The reason for it's need is because each query to submission time can takes around 6 to 100 microseconds and can add up when queried for internal dependency events as well

Reply:
> A user visible event is ...

At the very least you should explain this in a source code comment. Although I am still not convinced that we need to extend PI API with this.

> The reason for it's need is because each query to submission time can takes around 6 to 100 microseconds and can add up when queried for internal dependency events as well

What is the case that is showing critical performance degradation without such PI API extension?
Aren't only those events are queried that user/app explicitly asked for?

Reply:
> Aren't only those events are queried that user/app explicitly asked for?
If there's a case where the command group submitted to queue.submit has memory transfers to and/or from the device then the events associated with those operations are also queried.

> What is the case that is showing critical performance degradation without such PI API extension?
I might have made a lapse in judgement. I thought that theoretically those degradation should add up to be significant but after testing., it's actually better to not use the PI API extension

Reply:
Changes applied